actions/ql/src/Security/CWE-829/UnpinnedActionsTag.ql

```diff
@@ -34,8 +40,19 @@ where
   ) and
   uses.getVersion() = version and
   not isTrustedOwner(nwo) and
-  not isPinnedCommit(version) and
-  not isImmutableAction(uses, nwo)
+  not isImmutableAction(uses, nwo) and
+  not
+  (
+    (
+      isContainerImage(nwo) and
+      isPinnedContainer(version)
+    )
+    or
+    (
+      not isContainerImage(nwo) and
+      isPinnedCommit(version)
+    )
+  )
```

Comment:
Sorry for the drive-by comment, but it's rare to see a place where you could actually use an [`if-then-else`](https://codeql.github.com/docs/ql-language-reference/formulas/#if-then-else) instead! 🙂
Specifically
```ql
if isContainerImage(nwo)
then isPinnedContainer(version)
else isPinnedCommit(version)
```